src/promptflow/tests/sdk_cli_azure_test/unittests/test_run_operations.py

```diff
@@ -0,0 +1,18 @@
+import tempfile
+
+import pytest
+from pytest_mock import MockerFixture
+
+from promptflow.azure import PFClient
+from promptflow.exceptions import UserErrorException
+
+
+@pytest.mark.unittest
+class TestRunOperations:
+    def test_download_run_with_invalid_workspace_datastore(self, pf: PFClient, mocker: MockerFixture):
+        # test download with invalid workspace datastore
+        mocker.patch.object(pf.runs, "_validate_for_run_download")
+        mocker.patch.object(pf.runs, "_workspace_default_datastore", "test")
+        with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
pytest has fixtures for temp file/folder: <https://docs.pytest.org/en/6.2.x/tmpdir.html>

so next time you might not need `tempfile` anymore

Reply:
Thanks, good to know it.